packages/lesswrong/components/form-components/TagSelect.tsx

```diff
@@ -16,48 +16,42 @@ const TagSelectStyles = (theme: ThemeType): JssStyles => ({
   },
 });
 
-const TagSelect = ({ value, path, document, classes, label, updateCurrentValues }: {
+const TagSelect = ({value, path, classes, label, updateCurrentValues}: {
   value: string,
   path: string,
   document: any,
   classes: ClassesType,
   label?: string,
   updateCurrentValues<T extends {}>(values: T): void,
 }) => {
-  const [currentId, setCurrentId] = useState(document.parentTag?._id);
-  const { document: parentTag, loading } = useSingle({
-    skip: !currentId,
-    documentId: currentId,
+  const {document: selectedTag, loading} = useSingle({
+    skip: !value,
+    documentId: value,
     collectionName: "Tags",
     fragmentName: 'TagBasicInfo',
   });
 
-  useEffect(() => {
-    // updateCurrentValues needs to be called after loading the TagBasicInfo query because
-    // when the query returns `value` gets set back to undefined for some reason. I think this
-    // is probably because it updates local storage somehow, but I'm not sure. This fixes it anyway
```

Comment:
This component is currently only used in one place - on the tag edit page to select the parent tag. Most of the component is just extra logic trying to work around the bug described in this comment, but it also hardcodes `document.parentTag?._id` which means it can't be used generically in vulcan forms.

The changes here fix the bug (which was just caused by only including `parentTag` and not `parentTagId` in the tag edit fragment), and also remove the hard-coded references to `parentTag` so we can use it in other places.